### PR TITLE
add missing vibration on the button 

### PR DESCRIPTION
**Pull request by Amitkumarkoli**
pr #117 
Hey! I just add the missing vibration on intro page and ready_for_session_page .
let me know is this correct or not?

### CODE REVIEW

File: lib/presentation/pages/intro_page.dart [hunk 1]
```diff
@@ -22,7 +24,16 @@ class _IntroPageState extends State<IntroPage> {
   IntroState state = IntroState.welcome;
   final Duration selectionTransitionDelay = const Duration(milliseconds: 200);
 
-  void nextPage() => _introKey.currentState?.next();
+   Future<void> _vibrateDevice() async {
+    if (await Vibrate.canVibrate) {
+      Vibrate.vibrate();
+    }
```
Comment (guyluz11) [inline lib/presentation/pages/intro_page.dart:30]: We are using controllers for api calls.
Use     `VibrationController.instance.vibrate(VibrationType.light);` directly

**Review by guyluz11 — CHANGES_REQUESTED** [review]
Welcome to Infinite Horizons.

I have added request for change, please fix and press the re review button for me to start checking the pr again.

**Comment by Amitkumarkoli** [comment]
Hey! I added new changes to it 2 days ago, 
can u please check it and let me know about this?

File: lib/presentation/pages/intro_page.dart [hunk 2]
```diff
@@ -22,10 +23,16 @@ class _IntroPageState extends State<IntroPage> {
...
+    _introKey.currentState?.next();
+  }
 
-  void onDone(BuildContext context) => Navigator.of(context)
-      .push(MaterialPageRoute(builder: (context) => HomePage()));
+  void onDone(BuildContext context) {
+    VibrationController.instance.vibrate(VibrationType.light);
+    Navigator.of(context)
+        .push(MaterialPageRoute(builder: (context) => HomePage()));
+  }
```
Comment (guyluz11) [inline lib/presentation/pages/intro_page.dart:35]: Not needed as you already called vibration on `ready_for_session_page.dart` which call this method

File: lib/presentation/pages/ready_for_session_page.dart [hunk 3]
```diff
@@ -35,7 +36,10 @@ class ReadyForSessionPage extends StatelessWidget {
       scaffold: false,
       title: 'start_session',
       child: ReadyForSessionOrganism(
-        callback,
+        () {
+          VibrationController.instance.vibrate(VibrationType.light);
+          callback();
+        },
```
Comment (guyluz11) [inline lib/presentation/pages/ready_for_session_page.dart:42]: Would be nicer to extract that to a method

**Review by guyluz11 — CHANGES_REQUESTED** [review]
Sorry for the late response, not sure how did I missed it.

**Comment by Amitkumarkoli** [comment]
Hey ! I just added new commit let me know this is what u want ?

File: lib/presentation/pages/intro_page.dart [hunk 4]
```diff
@@ -22,10 +23,15 @@ class _IntroPageState extends State<IntroPage> {
...
+    VibrationController.instance.vibrate(VibrationType.light);
+    _introKey.currentState?.next();
+  }
 
-  void onDone(BuildContext context) => Navigator.of(context)
-      .push(MaterialPageRoute(builder: (context) => HomePage()));
+  void onDone(BuildContext context) {
+    Navigator.of(context)
+        .push(MaterialPageRoute(builder: (context) => HomePage()));
+  }
```
Comment (guyluz11) [inline lib/presentation/pages/intro_page.dart:34]: Please revert to arrow function

File: lib/presentation/pages/ready_for_session_page.dart [hunk 5]
```diff
@@ -31,13 +32,20 @@ class ReadyForSessionPage extends StatelessWidget {
...
       ),
     );
   }
+
+  void handleCallback() => _handleCallback();
+
+  void _handleCallback() {
+    VibrationController.instance.vibrate(VibrationType.light);
+    callback();
+  }
```
Comment (guyluz11) [inline lib/presentation/pages/ready_for_session_page.dart:50]: There is no reason to call function to call function??.

Also move it above the `build` method

File: lib/presentation/pages/ready_for_session_page.dart [hunk 6]
```diff
@@ -31,13 +32,20 @@ class ReadyForSessionPage extends StatelessWidget {
         text = "So much energy 🔋⚡🔋⚡🔋⚡🔋\nLet's begin";
     }
 
-    return PageEnclosureMolecule(
+     return PageEnclosureMolecule(
       scaffold: false,
       title: 'start_session',
       child: ReadyForSessionOrganism(
-        callback,
+        () => handleCallback(),
```
Comment (guyluz11) [inline lib/presentation/pages/ready_for_session_page.dart:39]: You can write this code cleaner by changing  `() => handleCallback(),`  to just `handleCallback`

**Review by guyluz11 — CHANGES_REQUESTED** [review]

**Review by guyluz11 — CHANGES_REQUESTED** [review]
Also this pr have conflicts with dev branch, please fix it as well

**Pull request closed without merging**